yeonjae/programmers-0/OX퀴즈.js

```diff
@@ -0,0 +1,21 @@
+function solution(quiz) {
+  const answer = [];
+  let result;
+
+  for (let i = 0; i < quiz.length; i++) {
+    const newArr = quiz[i].split(" ");
+    if (newArr[1] === "-") {
+      result =
+        parseInt(newArr[0]) - parseInt(newArr[2]) === parseInt(newArr[4])
+          ? "O"
+          : "X";
+    } else {
+      result =
+        parseInt(newArr[0]) + parseInt(newArr[2]) === parseInt(newArr[4])
+          ? "O"
+          : "X";
+    }
+    answer.push(result);
+  }
+  return answer;
+}
```

Comment:
```suggestion
function solution(quiz) {
  const answer = [];

  for (let i = 0; i < quiz.length; i++) {
    const [operand1, operator, operand2, equalSign, formulaResult] = quiz[i]
      .split(" ")
      .map((el) => (isNaN(+el) ? el : +el));

    if (operator === "-") {
      answer.push(operand1 - operand2 === formulaResult ? "O" : "X");
    } else {
      answer.push(operand1 + operand2 === formulaResult ? "O" : "X");
    }
  }
  return answer;
}
```

- 풀이 잘봤습니다~ 🙇‍♂️ 지금보면 반복적으로 `parseInt()`를 하는게 코드를 길게 복잡하게 만드는 것 같습니다! 아래와 같이 `map` + `구조분해 할당`을 이용해서 반복적인 작업을 개선할 수 있습니다. 이렇게 작업할 경우 newArr[0],[1],[2]와 같은 명확하지 않은 값을 사용하는게 아닌 operator, operand 등의 명확한 의미의 변수로 활용할 수 있습니다.
- 또한 지역변수 result는 불필요한 변수입니다. 만약 사용하더라도 for문 내에 선언하셔서 사용하셔야 됩니다.
- 아래 코드는 map을 활용해서 곧바로 return한 코드입니다. 
```js
function solution(quiz) {
  return quiz.map((el) => {
    const [operand1, operator, operand2, equalSign, formulaResult] = el
      .split(" ")
      .map((el) => (isNaN(+el) ? el : +el));

    const sum = operator === "-" ? operand1 + operand2 * -1 : operand1 + operand2;

    return sum === formulaResult ? "O" : "X";
  });
}
```

Reply:
아 map 안에서 저렇게 구조분해할당으로 반복적인 코드를 줄일수있군요?? 이런식으로 써볼생각은 못했는데 기억해두겠습니다!